example/App.tsx

```diff
@@ -1,71 +1,93 @@
-import { useEffect } from 'react';
-import { Button, StyleSheet, Text, View } from 'react-native';
+import { useEffect, useState } from "react";
+import { Button, StyleSheet, Text, View } from "react-native";
 
-import * as SiroReactNative from 'siro-react-native';
+import * as SiroReactNative from "siro-react-native";
 
 const interaction: SiroReactNative.Interaction = {
-  id: '123',
+  id: "123",
 
-  userId: 'externalId',
-  note: 'Got the sale!',
+  userId: "externalId",
+  note: "Got the sale!",
   leadDateCreated: new Date(),
   dateCreated: new Date(),
   contacts: [],
-  leadId: 'leadId',
+  leadId: "leadId",
   coordinates: { latitude: 0, longitude: 0 },
   stage: {
-    id: 'stageId',
-    name: 'stageName',
-    color: '#000',
-    icon: 'icon.jpg',
+    id: "stageId",
+    name: "stageName",
+    color: "#000",
+    icon: "icon.jpg",
     won: true,
     interacted: true,
   },
   metadata: {
-    some: 'data',
+    some: "data",
     someNumber: 5,
     somebool: true,
-    someArray: ['rrr', 2, true],
-  }
-}
+    someArray: ["rrr", 2, true],
+  },
+};
 
 export default function App() {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+
   useEffect(() => {
-    setupSiro()
-  }, [])
+    setupSiro();
+  }, []);
 
   const setupSiro = () => {
     SiroReactNative.setup(SiroReactNative.Environment.staging);
-  }
+  };
 
   const startRecording = () => {
     SiroReactNative.startRecording();
-  }
+  };
 
   const stopRecording = () => {
     SiroReactNative.stopRecording();
-  }
+  };
 
   const sendEvent = (eventName: string, data?: SiroReactNative.Interaction) => {
     SiroReactNative.sendEvent(eventName, data);
-  }
+  };
 
   const showModal = () => {
     SiroReactNative.showModal();
-  }
+  };
+
+  setInterval(() => {
+    setIsLoggedIn(SiroReactNative.isUserLoggedIn());
```

Comment:
I'm not great at react, but I imagine there has to be a better way to react to changes in isUserLoggedIn than using an interval?

Reply:
yeah there is, usually we can do that using something like a `useEffect`, the problem here is that since the login data is coming from Swift, it makes it a bit more complicated.

The ideal solution when it comes to React Native + Native modules is to use events to notify the RN app of the change in user login state. That's something I def want to add for a future update. For this version Gamify agreed that they can set up an interval/watcher to watch the login state so we can launch sooner.